internal/server/handler.go

```diff
@@ -96,10 +96,14 @@ func (s *Server) handler() http.HandlerFunc {
 		}
 
 		// Need to sign the header, just before sending it
-		r = signer.SignV4(*r, s.cfg.S3Settings.AccessKeyID,
-			s.cfg.S3Settings.SecretAccessKey,
-			val.SessionToken,
-			s.cfg.S3Settings.Region)
+		if s.cfg.S3Settings.AccessKeyID == "" && s.cfg.S3Settings.SecretAccessKey == "" {
```

Comment:
It seems like this `if` condition is repeated in two places. Please create a helper method for server struct and refactor it there.

Reply:
Thanks @agnivade I added some changes.